code/modules/vending/vending.dm

```diff
@@ -3372,13 +3374,41 @@ ABSTRACT_TYPE(/obj/machinery/vending/jobclothing)
 	New()
 		..()
 		START_TRACKING
+		src.fill_image = image(src.icon, "medchem-0", -1)
+		src.fill_image.plane = PLANE_ABOVE_LIGHTING
+		src.UpdateIcon()
 
 	disposing()
 		. = ..()
 		STOP_TRACKING
 
+	update_icon()
+		..()
+		if (status & (BROKEN|NOPOWER))
+			src.fill_image.icon_state = null
+			src.UpdateOverlays(src.fill_image, "fill_image")
+			return
+		var/total_reagents = 0
+		var/fluid_state = 0
+		for(var/datum/data/vending_product/player_product/R in src.player_list)
+			for(var/obj/item/product in R.contents)
+				if(istype(product, /obj/item/item_box/medical_patches) || istype(product, /obj/item/chem_pill_bottle))
+					for(var/obj/item/reagent_containers/container in product)
+						total_reagents += container.reagents.total_volume
+				if(istype(product, /obj/item/reagent_containers))
+					var/obj/item/reagent_containers/container = product
+					total_reagents += container.reagents.total_volume
```

Comment:
You could probably cut the istypes here since it can be assumed that we want to read the reagents and contents of anything we put in here

Reply:
yup did this and it's much cleaner and easier to read and works!